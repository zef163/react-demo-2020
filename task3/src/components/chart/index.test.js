import {render, screen} from '@testing-library/react';
import * as redux from 'react-redux'

import Chart from './'

let useSelector;

beforeAll(() => {
    useSelector = jest.spyOn(redux, 'useSelector');
});

afterAll(() => {
    useSelector.mockRestore();
})

jest.mock('components/button', () => ({
    ButtonWrapper: 'ButtonWrapper',
    Button: 'Button'
}));
jest.mock('components/lineChart', () => ({
    __esModule: true,
    default: 'LineChart',
}))

describe('<Chart />', () => {
    it('should render', () => {
        useSelector.mockReturnValue({
            data: [{date: '2020-01-01', val: 1}, {date: '2020-01-02', val: 2}],
            startDate: '2020-01-01',
            rangeType: 'week',
        });
        expect(render(<Chart changeRange={jest.fn()} />)).toMatchSnapshot();
    });

    it.each(['Week', 'Month', 'Quarter', 'Year', 'Max'])(
        'should call "changeRange" from props (%s)',
        (text) => {
            useSelector.mockReturnValue({
                data: [],
                startDate: '2020-01-01',
                rangeType: 'week',
            });
            const changeRange = jest.fn();

            render(<Chart changeRange={changeRange} />);
            const button = screen.getByText(text);

            expect(changeRange).not.toBeCalled();
            button.click();
            expect(changeRange).toHaveBeenCalledWith(text.toLowerCase());
            expect(changeRange).toHaveBeenCalledTimes(1);
        }
    );
});