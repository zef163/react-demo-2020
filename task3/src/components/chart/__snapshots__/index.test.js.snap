// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chart /> should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <div
          class="header"
        >
          <div
            class="title"
          >
            NII CAPITAL 7.625 21
            <span
              class="currency"
            >
              USD
            </span>
          </div>
          <div
            class="info"
          >
            US67021BAE92
            <br />
            NII CAPITAL CORP, Telecommunications, NR, till 
            01.01.2020
          </div>
          <hr
            class="separator"
          />
          <buttonwrapper>
            <button>
              Week
            </button>
            <button>
              Month
            </button>
            <button>
              Quarter
            </button>
            <button>
              Year
            </button>
            <button>
              Max
            </button>
          </buttonwrapper>
        </div>
        <linechart
          data="[object Object],[object Object]"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <div
        class="header"
      >
        <div
          class="title"
        >
          NII CAPITAL 7.625 21
          <span
            class="currency"
          >
            USD
          </span>
        </div>
        <div
          class="info"
        >
          US67021BAE92
          <br />
          NII CAPITAL CORP, Telecommunications, NR, till 
          01.01.2020
        </div>
        <hr
          class="separator"
        />
        <buttonwrapper>
          <button>
            Week
          </button>
          <button>
            Month
          </button>
          <button>
            Quarter
          </button>
          <button>
            Year
          </button>
          <button>
            Max
          </button>
        </buttonwrapper>
      </div>
      <linechart
        data="[object Object],[object Object]"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
